include/rellic/AST/IRToASTVisitor.h

```diff
@@ -26,70 +26,49 @@
 
 namespace rellic {
 
-using IRToTypeDeclMap = std::unordered_map<llvm::Type *, clang::TypeDecl *>;
-using IRToValDeclMap = std::unordered_map<llvm::Value *, clang::ValueDecl *>;
-using IRToStmtMap = std::unordered_map<llvm::Value *, clang::Stmt *>;
-using ArgToTempMap = std::unordered_map<llvm::Argument *, clang::VarDecl *>;
-
-class IRToASTVisitor : public llvm::InstVisitor<IRToASTVisitor, clang::Stmt *> {
+class IRToASTVisitor : public llvm::InstVisitor<IRToASTVisitor, clang::Expr *> {
  private:
   clang::ASTContext &ast_ctx;
 
   ASTBuilder ast;
 
-  IRToTypeDeclMap &type_decls;
-  IRToValDeclMap &value_decls;
-  StmtToIRMap &provenance;
-  ArgToTempMap &temp_decls;
+  Provenance &provenance;
   size_t num_literal_structs = 0;
   size_t num_declared_structs = 0;
 
   clang::QualType GetQualType(llvm::Type *type);
 
+  clang::Expr *CreateConstantExpr(llvm::Constant *constant);
   clang::Expr *CreateLiteralExpr(llvm::Constant *constant);
 
   clang::Decl *GetOrCreateIntrinsic(llvm::InlineAsm *val);
 
-  clang::Expr *GetTempAssign(llvm::Instruction &inst, clang::Expr *expr);
-
  public:
-  IRToASTVisitor(StmtToIRMap &provenance, clang::ASTUnit &unit,
-                 IRToTypeDeclMap &type_decls, IRToValDeclMap &value_decls,
-                 IRToStmtMap &stmts, ArgToTempMap &temp_decls);
+  IRToASTVisitor(clang::ASTUnit &unit, Provenance &provenance);
 
-  clang::Stmt *GetOrCreateStmt(llvm::Value *val);
   clang::Decl *GetOrCreateDecl(llvm::Value *val);
-  clang::Expr *GetOperandExpr(llvm::Value *val);
-
-  StmtToIRMap &GetStmtToIRMap() { return provenance; }
-  IRToValDeclMap &GetIRToValDeclMap() { return value_decls; }
-  IRToTypeDeclMap &GetIRToTypeDeclMap() { return type_decls; }
+  clang::Expr *GetOperandExpr(llvm::Use &val);
```

Comment:
Probably should be renamed to `CreateOperandExpr(llvm::Use &val)` since it no longer "gets" anything from a map.